### ISSUE TITLE
Bug in build-json-js for prose.see_also

### ISSUE THREAD

**Issue by peterbe** [mdn/stumptown-experiment#32]
`prose.see_also` is [required for html-element recipes](https://github.com/mdn/stumptown-experiment/blob/2c5a398d67af222596fee30361269207bc1acc24/recipes/html-element.yaml#L16) but when I generate `packaged/html/elements/video.json` it's not there in the JSON.

**Comment by peterbe** [comment]
Instead of fixing this the straight-forward way first, I'd be interested in taking this opportunity to write a validation script that *would have* caught this. It'd effectively be an integration test. Something like:

```
for element in $(ls content/html/elements) {
  ./scripts/build-json.js $element
  ./scripts/validate-built-json.js packaged/$element.json --recipe=recipes/html-element.yaml
}
```
It doesn't have to be fancy but definitely useful to have to achieve that perfect integrity. 

It would also help catch things like someone accidentally deleting the `<!-- see_also -->\ntext text text` in a PR when they edit the `prose.md` file.

**Comment by wbamberg** [comment]
I believe this is a bug in the source, not the builder. The bit of prose.md is like this:

```
Captions should not obstruct the main subject of the video. They can be
positioned using [the `align` VTT cue
setting](https://developer.mozilla.org/en-US/docs/Web/API/WebVTT_API#Cue_settings).

- [MDN Subtitles and closed caption —
  Plugins](https://developer.mozilla.org/en-US/docs/Plugins/Flash_to_HTML5/Video/Subtitles_captions)
- [Web Video Text Tracks Format
  (WebVTT)](https://developer.mozilla.org/en-US/docs/Web/API/WebVTT_API)
- [WebAIM: Captions, Transcripts, and Audio
  Descriptions](https://webaim.org/techniques/captions/)
- [MDN Understanding WCAG, Guideline 1.2
  explanations](https://developer.mozilla.org/en-US/docs/Web/Accessibility/Understanding_WCAG/Perceivable#Guideline_1.2_—_Providing_text_alternatives_for_time-based_media)
- [Understanding Success Criterion 1.2.1 | W3C Understanding WCAG
  2.0](https://www.w3.org/TR/UNDERSTANDING-WCAG20/media-equiv-av-only-alt.html)
- [Understanding Success Criterion 1.2.2 | W3C Understanding WCAG
  2.0](https://www.w3.org/TR/UNDERSTANDING-WCAG20/media-equiv-captions.html)
<!-- see-also -->
See also
--------

- [Media formats supported by the audio and video
  elements](https://developer.mozilla.org/en-US/docs/Media_formats_supported_by_the_audio_and_video_elements)
```

Because there isn't a blank line between the "Understanding Success Criterion 1.2.2" item and the comment, then Markdown thinks the comment is part of the list item, so the HTML looks like:

```html
<p>Captions should not obstruct the main subject of the video. They can be
positioned using <a href="https://developer.mozilla.org/en-US/docs/Web/API/WebVTT_API#Cue_settings">the <code>align</code> VTT cue
setting</a>.</p>
<ul>
<li><a href="https://developer.mozilla.org/en-US/docs/Plugins/Flash_to_HTML5/Video/Subtitles_captions">MDN Subtitles and closed caption —
Plugins</a></li>
<li><a href="https://developer.mozilla.org/en-US/docs/Web/API/WebVTT_API">Web Video Text Tracks Format
(WebVTT)</a></li>
<li><a href="https://webaim.org/techniques/captions/">WebAIM: Captions, Transcripts, and Audio
Descriptions</a></li>
<li><a href="https://developer.mozilla.org/en-US/docs/Web/Accessibility/Understanding_WCAG/Perceivable#Guideline_1.2_%E2%80%94_Providing_text_alternatives_for_time-based_media">MDN Understanding WCAG, Guideline 1.2
explanations</a></li>
<li><a href="https://www.w3.org/TR/UNDERSTANDING-WCAG20/media-equiv-av-only-alt.html">Understanding Success Criterion 1.2.1 | W3C Understanding WCAG
2.0</a></li>
<li><a href="https://www.w3.org/TR/UNDERSTANDING-WCAG20/media-equiv-captions.html">Understanding Success Criterion 1.2.2 | W3C Understanding WCAG
2.0</a><!-- see-also -->
See also</li>
</ul>
<hr>
<ul>
<li><a href="https://developer.mozilla.org/en-US/docs/Media_formats_supported_by_the_audio_and_video_elements">Media formats supported by the audio and video
elements</a></li>
```

The JSON builder then sweeps that comment up when it consumes the list, which it considers part of the previous section ("accessibility_concerns"), so it doesn't see it as a section delimiter.

I think the correct thing to do here is to add a blank line before the comment, then it will be processed properly. This is the kind of thing that makes me worried about Markdown as a source format...

As for tests: a linter would catch that "see-also" is missing and throw an error because "see-also" is mandatory, and that would indirectly help us find the problem. Unfortunately I'm not sure how we could test that the Markdown is properly formatted here.

**Comment by peterbe** [comment]
> a linter would catch that "see-also" is missing and throw an error

Yes! That's why we need a linter that can combine the recipes with the elements (e.g HTML video). More reason to keep the recipes here with the content I suppose. 

> Unfortunately I'm not sure how we could test that the Markdown is properly formatted here.

We could have a custom regex like `/\n\n<!-- \w+ -->/` or something fancy (and fragile).

Another option is to run every .md file through `markdownlint`
For example, I pasted the (before this was merged) `prose.md` into https://dlaa.me/markdownlint/ and it correctly spotted that the...

```
See also
----
```
was lacking a blank line. 
```
MD022 Headings should be surrounded by blank lines [Expected: 1; Actual: 0; Above] [Context: "<!--          -->"]
```

I haven't been able to keep up on the markdown linting stuff. I know that https://github.com/DavidAnson/markdownlint is pretty mature. It would be cool to use that to make sure every single .md file has exactly 0 warnings from `markdownlint`.
